packages/react-compose/src/types.ts

```diff
@@ -109,6 +125,9 @@ export type ComposePreparedOptions<TProps = {}, TState = TProps> = {
   slots: Record<string, React.ElementType> & { __self: React.ElementType };
   slotProps: ((props: TProps) => Record<string, object>)[];
 
+  // tslint:disable-next-line:no-any
+  state: (props: TState, options: ComposePreparedOptions) => any;
```

Comment:
shouldn't the typing be something like:
`(props: TParentState, options: ComposePreparedOptions) => TInputState;`

Reply:
actually, don't we need to also give  inputProps?
so something like:
`(props: TInputProps, parentState: TParentState | undefined, options: ComposePreparedOptions) => TInputState;`